src/Avalonia.Controls/Utils/BorderRenderHelper.cs

```diff
@@ -84,17 +92,26 @@ void Update(Size finalSize, Thickness borderThickness, CornerRadius cornerRadius
 
         public void Render(DrawingContext context,
             Size finalSize, Thickness borderThickness, CornerRadius cornerRadius,
-            IBrush background, IBrush borderBrush, BoxShadows boxShadows)
+            IBrush background, IBrush borderBrush, BoxShadows boxShadows, double borderDashOffset = 0,
+            PenLineCap borderLineCap = PenLineCap.Flat, PenLineJoin borderLineJoin = PenLineJoin.Miter,
+            AvaloniaList<double> borderDashArray = null)
         {
             if (_size != finalSize
                 || _borderThickness != borderThickness
                 || _cornerRadius != cornerRadius
+                || _borderLineCap != borderLineCap
```

Comment:
We don't need to update the geometry etc. when those new properties change. They don't affect measurements. Please remove them from this check.

Reply:
We always pass current values to RenderCore so the render path should always use updated values.

Reply:
Yes you are right, thank you